### PR TITLE
Zer 83

### PR DESCRIPTION
**Pull request by Nazarii-Makohin**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by halyna-m — APPROVED** [review]

**Review by andriibureviy — DISMISSED** [review]

**Review by halyna-m — APPROVED** [review]

File: app/models/concerns/omniauth_attributes_concern.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module OmniauthAttributesConcern
+  extend ActiveSupport::Concern
+  module ClassMethods
+    def twitter(params)
```
Comment (maksymst) [inline app/models/concerns/omniauth_attributes_concern.rb:6]: twitter?

File: app/models/user.rb [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+
+  has_many :user_authentications
+
   # Include default devise modules. Others available are:
   #  :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable,
          :lockable, :timeoutable, :trackable, :confirmable,
          :omniauthable, omniauth_providers: %i[google_oauth2]
+  devise :omniauthable, omniauth_providers: %i[facebook]
```
Comment (maksymst) [inline app/models/user.rb:14]: looks at the previous line

File: app/models/user.rb [hunk 3]
```diff
@@ -19,12 +24,24 @@ class User < ApplicationRecord
...
       user.password = Devise.friendly_token[0, 20]
       user.first_name = auth.info.first_name
     end
+
+    name_split = auth.info.name.split
+    user = User.where(email: auth.info.email).first
+    user ||= User.create!(provider: auth.provider, uid: auth.uid,
+                          last_name: name_split[0], first_name: name_split[1],
+                          email: auth.info.email,
+                          password: Devise.friendly_token[0, 20])
```
Comment (maksymst) [inline app/models/user.rb:43]: you already have a user creating in the method. why do you need this one?

File: app/views/devise/shared/_links.html.erb [hunk 4]
```diff
@@ -16,6 +16,7 @@
       <%= link_to "Sign in with Google", omniauth_authorize_path(resource_name, provider) %><br />
     <%- else %>
       <%= link_to "Sign in with #{OmniAuth::Utils.camelize(provider)}", omniauth_authorize_path(resource_name, provider) %><br />
+      <%= link_to "Sign in with Google", omniauth_authorize_path(resource_name, provider) %><br />
```
Comment (maksymst) [inline app/views/devise/shared/_links.html.erb:19]: why do you have google sign in link here?

File: config/webpacker.yml [hunk 5]
```diff
@@ -90,27 +90,3 @@ production:
 
   # Cache manifest.json for performance
   cache_manifest: true
-
```
Comment (maksymst) [inline config/webpacker.yml:93]: why did you remove this code?

File: config/webpack/environment.js [hunk 6]
```diff
@@ -1,12 +1,3 @@
 const { environment } = require('@rails/webpacker')
-const webpack = require('webpack')
-
-environment.plugins.append('Provide',
-  new webpack.ProvidePlugin({
-    $: 'jquery',
-    jQuery: 'jquery',
-    Popper: ['popper.js', 'default']
-  })
-)
```
Comment (maksymst) [inline config/webpack/environment.js:10]: why did you remove this code?

File: bin/webpack-dev-server [hunk 7]
```diff
@@ -1,19 +1,18 @@
...
-require 'bundler/setup'
+require "bundler/setup"
 
-require 'webpacker'
-require 'webpacker/dev_server_runner'
+require "webpacker"
+require "webpacker/dev_server_runner"
 
-APP_ROOT = File.expand_path('..', __dir__)
+APP_ROOT = File.expand_path("..", __dir__)
```
Comment (maksymst) [inline bin/webpack-dev-server:15]: what the purpose of this change?

File: bin/webpack [hunk 8]
```diff
@@ -1,19 +1,18 @@
...
-require 'bundler/setup'
+require "bundler/setup"
 
-require 'webpacker'
-require 'webpacker/webpack_runner'
+require "webpacker"
+require "webpacker/webpack_runner"
 
-APP_ROOT = File.expand_path('..', __dir__)
+APP_ROOT = File.expand_path("..", __dir__)
```
Comment (maksymst) [inline bin/webpack:15]: what is the purpose of this change?

**Review by maksymst — CHANGES_REQUESTED** [review]

File: app/views/devise/shared/_links.html.erb (hunk 4 above)
Comment (Nazarii-Makohin) [reply app/views/devise/shared/_links.html.erb:19]: without this, there will be only "Sign in with Facebook"
image

Comment (maksymst) [reply app/views/devise/shared/_links.html.erb:19]: yeah, because you made a mistake in User model
